__tests__/connection.test.ts

```diff
@@ -39,12 +39,6 @@ describe("Check canConnect", () => {
           true
         );
       });
-
-      it("canConnect failure", () => {
-        expect(this.canConnect(this.destination.in, this.source.out)).toBe(
```

Comment:
Leaving a marker for @axmmisaka to review.

Reply:
I don't recommend deleting it, but instead setting this to expecting `true` instead. 
What I am more worried about is that this is directly related to nothing but https://github.com/lf-lang/reactor-ts/pull/186/files#r1275393168, where we change from always disallowing `in` to `out` connection to always allowing.
If there's a FIXME, I recommend documenting what's wrong in an issue, and write relevant tests in a new PR that shows the expected behaviours (which might be failing, but it's okay, because we need to address the FIXME).
But just don't remove these tests because they are failing.

```mermaid
    graph
     0[/"testApp[M0]"/]
     subgraph "testApp.fc5128bb"
      1[/"testApp.fc5128bb.OutPort"\]
      2[/"testApp.fc5128bb[M0]"/]
     end
     subgraph "testApp.9d6363af"
      3[\"testApp.9d6363af.InPort"/]
      4[\"testApp.9d6363af.InPort"/]
      5[/"testApp.9d6363af[M0]"/]
     end
    
    0 --> 2
    0 --> 5
    3 --x|Allow this????| 1
```

Reply:
Thanks to @axmmisaka, I found that 
https://github.com/lf-lang/reactor-ts/blob/14f5b1352723185a1c8d5834168883ad380f332c/src/core/reactor.ts#L1205
this line cannot be simply turned into `true`. 

In the commit [01624c4](https://github.com/lf-lang/reactor-ts/pull/186/commits/01624c4a842451937a4e265ef8f7d61eb9944ee9), the connection between input and output ports are allowed only if those are in the same container and the container calls `canConnect` to connect them.
https://github.com/lf-lang/reactor-ts/blob/14965ac762988c2608913feaa8b6e262fc7dddab/src/core/reactor.ts#L1204-L1213
Also, I added a test related to this.
https://github.com/lf-lang/reactor-ts/blob/14965ac762988c2608913feaa8b6e262fc7dddab/__tests__/hierarchy.ts#L43-L48